app/javascript/packages/document-capture/context/acuant.tsx

```diff
@@ -0,0 +1,270 @@
+  onSuccess: () => void;
+  onFail: (code: AcuantInitializeCode, description: string) => void;
+}
+
+type AcuantInitialize = (
+  credentials: string | null,
+  endpoint: string | null,
+  callbackOptions?: AcuantCallbackOptions,
+) => void;
+
+type AcuantWorkersInitialize = (callback: () => void) => void;
+
+declare global {
+  interface AcuantJavascriptWebSdkInterface {
+    initialize: AcuantInitialize;
+    startWorkers: AcuantWorkersInitialize;
+    START_FAIL_CODE: string;
+    REPEAT_FAIL_CODE: string;
+    SEQUENCE_BREAK_CODE: string;
+  }
+}
+
+/**
+ * Start liveness capture
+ */
+type AcuantStartSelfieCapture = (callback: (nextImageData: string) => void) => void;
+interface AcuantPassiveLivenessInterface {
+  startSelfieCapture: AcuantStartSelfieCapture;
+}
+
+declare global {
+  interface Window {
+    /* Document load callback to assign Javascript Web SDK globals */
+    loadAcuantSdk: () => void;
+    /* Acuant configuration */
+    acuantConfig: AcuantConfig;
+    /* Acuant Passive Liveness API */
+    AcuantPassiveLiveness: AcuantPassiveLivenessInterface;
+    /* Possible AcuantJavascriptWebSdk on the window object (11.5.0) */
+    AcuantJavascriptWebSdk: AcuantJavascriptWebSdkInterface;
+    /* Possible AcuantCamera on the window object (11.5.0) */
+    AcuantCamera: AcuantCameraInterface;
+  }
+}
+
+interface AcuantContextProviderProps {
+  sdkSrc: string; // SDK source URL.
+  cameraSrc: string; // Camera JavaScript source URL.
+  credentials: string | null; // SDK credentials.
+  endpoint: string; // Endpoint to submit payload.
+  glareThreshold: number; // Minimum acceptable glare score for images.
+  sharpnessThreshold: number; // Minimum acceptable sharpness score for images.
+  children: ReactNode; // Child element
+}
+
+/**
+ * The minimum glare score value to be considered acceptable.
+ *
+ */
+export const DEFAULT_ACCEPTABLE_GLARE_SCORE = 30;
+
+/**
+ * The minimum sharpness score value to be considered acceptable.
+ *
+ */
+export const DEFAULT_ACCEPTABLE_SHARPNESS_SCORE = 30;
+
+/**
+ * Returns the containing directory of the given file, including a trailing slash.
+ */
+export const dirname = (file: string): string => {
+  return file.split('/').slice(0, -1).concat('').join('/');
+};
+
+interface AcuantContextInterface {
+  isReady: boolean;
+  isAcuantLoaded: boolean;
+  isError: boolean;
+  isCameraSupported: boolean | null;
+  isActive: boolean;
+  setIsActive: (nextIsActive: boolean) => void;
+  credentials: string | null;
+  glareThreshold: number;
+  sharpnessThreshold: number;
+  endpoint: string | null;
+}
+
+const AcuantContext = createContext<AcuantContextInterface>({
+  isReady: false,
+  isAcuantLoaded: false,
+  isError: false,
+  isCameraSupported: /** @type {boolean?} */ null,
+  isActive: false,
+  setIsActive: () => {},
+  credentials: null,
+  glareThreshold: DEFAULT_ACCEPTABLE_GLARE_SCORE,
+  sharpnessThreshold: DEFAULT_ACCEPTABLE_SHARPNESS_SCORE,
+  endpoint: /** @type {string?} */ null,
```

Comment:
These `/** @type */` annotations behave like a type-cast, but they don't work in `.tsx` files.

The equivalent is...


```suggestion
  isCameraSupported: null as boolean | null,
  isActive: false,
  setIsActive: () => {},
  credentials: null,
  glareThreshold: DEFAULT_ACCEPTABLE_GLARE_SCORE,
  sharpnessThreshold: DEFAULT_ACCEPTABLE_SHARPNESS_SCORE,
  endpoint: null as string | null,
```

Reply:
Updated in b563c26